### PR TITLE
Ansible ssh extra args

### LINKED ISSUE #9808: ansible-remote fails on windows due to identities only
https://github.com/hashicorp/packer/blob/master/provisioner/ansible/provisioner.go#L702-705
causes an issue on windows ssh

#### Overview of the Issue

adding identitiesonly=yes to the ansible-playbook command breaks the ability for windows ssh to connect with ansible
also It can not be overridden by simply adding the environment variable version to `ansible_env_vars`.

calling ansible-playbook without works

#### Reproduction Steps
use default windows ssh in your setup of build 
https://github.com/PowerShell/Win32-OpenSSH/wiki/Install-Win32-OpenSSH
use package build from below

### Packer version

From `packer version` 1.6.1

### Simplified Packer Buildfile

```
{
    "variables": {
        "iso_url": null,
        "iso_checksum": null,
        "identifier": null
    },
    "builders": [{
        "name": "windows-build",
        "type": "vmware-iso",
        "disk_size": 100000,
        "disk_type_id": "0",
        "guest_os_type": "windows9srv-64",
        "iso_url": "{{user `iso_url`}}",
        "iso_checksum": "sha256:{{user `iso_checksum`}}",
        "communicator": "ssh",
        "ssh_username": "{{env `ssh_username`}}",
        "ssh_password": "{{env `ssh_password`}}",
        "ssh_timeout": "240m",
        "ssh_handshake_attempts": 100,
        "ssh_read_write_timeout": "240m",
        "headless": "TRUE",
        "floppy_dirs": [
            "{{user `path`}}/floppy/{{user `identifier`}}/*"
        ],
        "shutdown_timeout": "120m",
        "cpus": 4,
        "cores": 1,
        "memory": 8192,
        "network": "nat",
        "network_adapter_type": "vmxnet3",
        "disk_adapter_type": "pvscsi",
        "version": "13",
        "keep_registered": false
    }],
    "provisioners": [{
        "playbook_file": "{{user `path`}}/template.yaml",
        "type": "ansible",
        "ansible_env_vars": ["SSH_EXTRA_ARGS=''"],

        "extra_arguments": ["-vv",
            "-T 1200",
            "--extra-vars",
            "ansible_user={{env `ssh_username`}}",
            "--extra-vars",
            "ansible_password={{env `ssh_password`}}"
        ],
        "groups": "windows",
        "inventory_file": "./inventory.yaml",
        "pause_before": "20s",
        "timeout": "2h"
    }]
}
```
### Operating system and Environment details

running packer on Centos7
against a windows 2019 and windows 2016 servers.

### Log Fragments and crash.log files

fatal: ###.###.###.###]: FAILED! => {"ansible_facts": {}, "changed": false, "failed_modules": {"setup": {"f
ailed": true, "module_stderr": "Warning: Permanently added '###.###.###.###' (ECDSA) to the list of known hosts.\r\nParameter format not correct - ;
\r\n", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", "rc": 1}}, "msg": "The following modules failed to exec
ute: setup\n"}

### PR DESCRIPTION
**Pull request by gamethis**
This is to address #9808

I have added an argument of `ansible_ssh_extra_args`
It defaults to '-o IdentitiesOnly=yes' if nothing is supplied

Need some help on understanding where to update the documentation for the pull request. 

If your PR resolves any open issue(s), please indicate them like this so they will be closed when your PR is merged:

Closes #9808

### CODE REVIEW

**Comment by gamethis** [comment]
@SwampDragons do you believe this needs a documentation update as well?

File: provisioner/ansible/provisioner.go [hunk 1]
```diff
@@ -89,6 +89,8 @@ type Config struct {
 	AnsibleEnvVars []string `mapstructure:"ansible_env_vars"`
 	// The playbook to be run by Ansible.
 	PlaybookFile string `mapstructure:"playbook_file" required:"true"`
+	// Specifies --ssh-extra-args on command line defaults to -o IdentitiesOnly=yes
```
Comment (SwampDragons) [inline provisioner/ansible/provisioner.go:92]: This code comment will get turned into the documentation automagically if you call `make generate` from inside the project root.

**Comment by SwampDragons** [comment]
You have this marked as a WIP so I'm not merging but from my perspective it looks good. Is there anything else you need from me before we merge?

**Comment by gamethis** [comment]
nope all good. thank you

**Review by azr — APPROVED** [review]

**Pull request merged**